File: 102. Binary Tree Level Order Traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        out = []
        if root == None: return out
        curr_level = [root]
        while curr_level:
            curr_nodes = []
            next_level = []
            for node in curr_level:
                curr_nodes.append(node.val)
                if node.left != None:
                    next_level.append(node.left)
                if node.right != None:
                    next_level.append(node.right)
            out.append(curr_nodes)
            curr_level = next_level
            
        return out

File: 102. Binary Tree Level Order Traversal/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
    var out [][]int
    if root == nil {
        return out
    }
    
    todo := []*TreeNode{}
    todo = append(todo, root)
    for len(todo) != 0 {
        next_todo := []*TreeNode{}
        all_null := true
        curr_level := []int{}
        for ii := 0; ii < len(todo); ii++ {
            if todo[ii] != nil {
                all_null = false
                curr_level = append(curr_level, todo[ii].Val)
                next_todo = append(next_todo, todo[ii].Left, todo[ii].Right)
            }
        }
        if all_null {
            return out
        }
        out = append(out, curr_level)
        todo = next_todo
    }
    return out
}
